pkg/services/cloudmigration/gmsclient/gms_client.go

```diff
@@ -148,8 +158,54 @@ func (c *gmsClientImpl) StartSnapshot(ctx context.Context, session cloudmigratio
 	return &result, nil
 }
 
-func (c *gmsClientImpl) GetSnapshotStatus(context.Context, cloudmigration.CloudMigrationSession, cloudmigration.CloudMigrationSnapshot) (*cloudmigration.CloudMigrationSnapshot, error) {
-	panic("not implemented")
+func (c *gmsClientImpl) GetSnapshotStatus(ctx context.Context, session cloudmigration.CloudMigrationSession, snapshot cloudmigration.CloudMigrationSnapshot) (*cloudmigration.GetSnapshotStatusResponse, error) {
+	c.getStatusMux.Lock()
+	defer c.getStatusMux.Unlock()
+	logger := c.log.FromContext(ctx)
+
+	// Ensure we can't send requests more frequently than supported
+	if !c.getStatusLastQueried.IsZero() && time.Since(c.getStatusLastQueried) < c.minGetStatusPollingPeriod {
+		logger.Debug("skipping query to Grafana Migration Service because this request was made too soon after the previous one")
+		return &cloudmigration.GetSnapshotStatusResponse{
+			State: cloudmigration.SnapshotStateUnknown,
+		}, nil
+	}
+
+	path := fmt.Sprintf("%s/api/v1/status/%s/status", buildBasePath(c.domain, session.ClusterSlug), snapshot.GMSSnapshotUID)
+
+	// Send the request to gms with the associated auth token
+	req, err := http.NewRequest(http.MethodGet, path, nil)
```

Comment:
Could we add a timeout?

Reply:
Done, added timeouts to the http clients for each method in here